src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -311,13 +321,42 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             return new FunctionDefinition(descriptor, instanceFactory, listenerFactory);
         }
 
-        private static FunctionDescriptor CreateFunctionDescriptor(MethodInfo method, string triggerParameterName,
+        // Expose internally for testing purposes 
+        internal static FunctionDescriptor FromMethod(
+            MethodInfo method, 
+            IJobActivator jobActivator = null,
+            INameResolver nameResolver = null)
+        {
+            var disabled = HostListenerFactory.IsDisabled(method, nameResolver, jobActivator);
+
+            // Determine the TraceLevel for this function (affecting both Console as well as Dashboard logging)
+            TraceLevelAttribute traceAttribute = TypeUtility.GetHierarchicalAttributeOrNull<TraceLevelAttribute>(method);
+
+            bool hasCancellationToken = method.GetParameters().Any(p => p.ParameterType == typeof(CancellationToken));
+        
+            return new FunctionDescriptor
+            {
+                Id = method.GetFullName(),
+                LogName = method.Name,
```

Comment:
out of curiosity, what's an example of `.Name` vs `.GetFullName` vs `.GetShortName` - edit, nvmd I see you mentioned that in the comments, nice!